solc/CommandLineInterface.cpp

```diff
@@ -2019,7 +1087,7 @@ void CommandLineInterface::outputCompilationResults()
 
 	if (
 		!m_compiler->compilationSuccessful() &&
-		m_stopAfter == CompilerStack::State::CompilationSuccessful
+		m_options.stopAfter == CompilerStack::State::CompilationSuccessful
 	)
```

Comment:
```suggestion
if (m_compiler->state() != m_options.stopAfter) ...
```
I think this should also work and is maybe quicker understandable than the composed if?

Reply:
I'll try. But this is something I'll have to test manually because it's not covered by tests.

Reply:
Actually I don't want to introduce breakage here so I think I'll do this one in a future, smaller PR where it can be properly reviewed. Here it would get lost in the noise.

Reply:
Also, if we don't cover it in the tests, maybe we want to add it then. Would that make sense?

Reply:
The problem here is that's in `CommandLineInterface.cpp` and most of that file remains hard to test (except maybe some static functions) because it just prints stuff to stdout. It would have to be refactored further to make the public functions return data in some structured form which then becomes the input for the functions that print the output. Basically, just like I extracted the parser, we should also extract the result formatter. I actually thought about refactoring it that way but then this PR would be even bigger so I dropped the idea for now.